src/modules/rest/controllers/rest.search.controller.ts

```diff
@@ -0,0 +1,151 @@
+import {Controller, Get, Inject, Optional, Query} from '@nestjs/common';
+import MongoOperationsService from '../../../services/mongo/mongo.operations.service';
+import {ApiOperation, ApiResponse, ApiTags} from '@nestjs/swagger';
+import {ApiImplicitQueries} from 'nestjs-swagger-api-implicit-queries-decorator';
+import {PAGINATION_IMPLICIT_QUERIES} from './common/rest.pagination.queries';
+import GlobalCharacterAggregateEntity from '../../data/entities/aggregate/global/global.character.aggregate.entity';
+import GlobalOutfitAggregateEntity from '../../data/entities/aggregate/global/global.outfit.aggregate.entity';
+import Pagination from '../../../services/mongo/pagination';
+import {SearchTermInterface} from '../../../interfaces/SearchTermInterface';
+
+@ApiTags('Search')
+@Controller('search')
+export default class RestSearchController {
+    constructor(
+        @Inject(MongoOperationsService) private readonly mongoOperationsService: MongoOperationsService,
+    ) {}
+
+    @Get()
+    @ApiOperation({summary: 'Searches GlobalCharacterAggregateEntity and GlobalOutfitAggregateEntity for a term'})
+    @ApiImplicitQueries([...PAGINATION_IMPLICIT_QUERIES, {
+        name: 'type',
+        required: false,
+        description: 'The type of the data to be searched, either "characters" or "outfits". If not specified, both types will be searched.',
+        type: String,
+        enum: ['characters', 'outfits'],
+    }])
+    @ApiResponse({
+        status: 200,
+        description: 'The list of GlobalCharacterAggregateEntity and GlobalOutfitAggregateEntity for a search term',
+        type: Object,
+        isArray: false,
+    })
+    async search(
+        @Query('searchTerm') searchTerm: string,
+        @Query('type') @Optional() type?: string,
```

Comment:
I would probably split this endpoint into two, one for characters and one for outfits. I also would use an actual enum as the type.

Also you might want to look at the NestJS Swagger CLI plugin.